src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/TraceEventExtensions.cs

```diff
@@ -23,6 +23,39 @@ public static bool TryGetCounterPayload(this TraceEvent traceEvent, CounterFilte
                 string series = payloadFields["Series"].ToString();
                 string counterName = payloadFields["Name"].ToString();
 
+                var metadataDict = new Dictionary<string, string>();
+
+                ReadOnlySpan<char> metadata = payloadFields["Metadata"].ToString();
```

Comment:
Can you make sure this works as expected for 3.1? From what I see in the code, it should but we won't exercise 3.1 much in our tooling.